packages/url_launcher/url_launcher/android/src/main/java/io/flutter/plugins/urllauncher/WebViewActivity.java

```diff
@@ -60,6 +64,48 @@ public boolean shouldOverrideUrlLoading(WebView view, WebResourceRequest request
 
   private IntentFilter closeIntentFilter = new IntentFilter(ACTION_CLOSE);
 
+  // Verifies that a url opened by `Window.open` has a secure url.
+  private class FlutterWebChromeClient extends WebChromeClient {
+    @Override
+    public boolean onCreateWindow(
+        final WebView view, boolean isDialog, boolean isUserGesture, Message resultMsg) {
+      final WebViewClient webViewClient =
+          new WebViewClient() {
+            @TargetApi(Build.VERSION_CODES.LOLLIPOP)
+            @Override
+            public boolean shouldOverrideUrlLoading(
+                @NonNull WebView view, @NonNull WebResourceRequest request) {
+              final String url = request.getUrl().toString();
+              if (isSecure(url)) {
+                webview.loadUrl(url);
+              }
+              return true;
+            }
+
+            @Override
+            public boolean shouldOverrideUrlLoading(WebView view, String url) {
+              if (isSecure(url)) {
+                webview.loadUrl(url);
+              }
+              return true;
+            }
+          };
+
+      final WebView newWebView = new WebView(webview.getContext());
+      newWebView.setWebViewClient(webViewClient);
+
+      final WebView.WebViewTransport transport = (WebView.WebViewTransport) resultMsg.obj;
+      transport.setWebView(newWebView);
+      resultMsg.sendToTarget();
+
+      return true;
+    }
+
+    private boolean isSecure(String url) {
+      return url.startsWith("https://") || url.startsWith("http://");
```

Comment:
I would suggest we rename this as: `isJavaScriptScheme(String url)` and check for `url.startsWith("javascript:")`. Link to https://tools.ietf.org/html/draft-hoehrmann-javascript-scheme-03 would be useful too.

Reply:
I would only do this after we get an ok from the internal team. This method is currently just following their suggestion.